internal/exec/engine.go

```diff
@@ -274,8 +276,21 @@ func (e *Engine) fetchReferencedConfig(cfgRef types.ConfigReference) (types.Conf
 	if err != nil {
 		return types.Config{}, err
 	}
+	headers := http.Header{}
+	for headerName, headerValue := range resource.ConfigHeaders {
+		headers.Add(headerName, headerValue[0])
+	}
+	if cfgRef.HTTPHeaders != nil && len(cfgRef.HTTPHeaders) > 0 {
+		if u.Scheme != "http" && u.Scheme != "https" {
+			return types.Config{}, e.Logger.Crit("Cannot use HTTP headers with %v source scheme", u.Scheme)
+		}
+		for _, headerData := range cfgRef.HTTPHeaders {
+			header := strings.Split(string(headerData), ":")
+			headers.Add(strings.TrimSpace(header[0]), strings.TrimSpace(header[1]))
```

Comment:
In case of malformed input (a string without `:`), this will panic with "index out of range". You might want to either recover the panic, or check the slice length first

Reply:
done